MapboxCoreNavigation/Dictionary.swift

```diff
@@ -0,0 +1,16 @@
+import Foundation
+
+extension Dictionary where Key == Int, Value: MGLStyleValue<NSNumber> {
+    /**
+     Returns a copy of the stop dictionary with each value multiplied by the given factor.
+     */
+    public func multiplied(by factor: Double) -> Dictionary {
+        var newCameraStop: [Int: MGLStyleValue<NSNumber>] = [:]
+        for stop in self as [Int : MGLStyleValue<NSNumber>] {
```

Comment:
Unsure if I can get away with not casting here.

Reply:
The only thing that comes to mind for me is that making a `private typealias NumericStyleDictionary = [Int: MGLStyleValue<NSNumber>]` (or something) might make the code a little cleaner.

Reply:
The overall extension requires the same type conformance in its `where` condition. If `as` builds without errors (as opposed to `as!` or `as?`), then casting is fine.

Reply:
No issues.